crates/syn-solidity/src/yul/stmt/switch.rs

```diff
@@ -0,0 +1,156 @@
+use crate::{kw, Spanned, YulBlock, YulExpr, YulLit};
+
+use proc_macro2::Span;
+use std::fmt;
+use syn::parse::{Parse, ParseStream, Result};
+
+/// A Yul switch statement can consist of only a default-case or one
+/// or more non-default cases optionally followed by a default-case.
+///
+/// Example switch statement in Yul:
+///
+/// ```solidity
+/// switch exponent
+/// case 0 { result := 1 }
+/// case 1 { result := base }
+/// default { revert(0, 0) }
+/// ```
+///
+/// Solidity Reference:
+/// <https://docs.soliditylang.org/en/latest/grammar.html#a4.SolidityParser.yulSwitchStatement>
+#[derive(Clone)]
+pub struct YulSwitch {
+    switch_token: kw::switch,
+    selector: YulExpr,
+    branches: Vec<YulCaseBranch>,
+    default_case: Option<YulSwitchDefault>,
+}
+
+impl Parse for YulSwitch {
+    fn parse(input: ParseStream<'_>) -> Result<Self> {
+        Ok(Self {
+            switch_token: input.parse()?,
+            selector: input.parse()?,
+            branches: {
+                let mut branches = vec![];
+                while input.peek(kw::case) {
+                    branches.push(input.parse()?);
+                }
+                branches
+            },
+            default_case: {
+                if input.peek(kw::default) {
+                    Some(input.parse()?)
+                } else {
+                    None
+                }
+            },
+        })
```

Comment:
from the spec looks like we need a default case if no other cases are provided, but here we may skip it. I think this is fine